lib/Frontend/ArgsToFrontendOutputsConverter.cpp

```diff
@@ -103,9 +100,10 @@ OutputFilesComputer::create(const llvm::opt::ArgList &args,
   ArrayRef<std::string> outputFileArguments =
       outputDirectoryArgument.empty() ? ArrayRef<std::string>(*outputArguments)
                                       : ArrayRef<std::string>();
-  const StringRef firstInput = inputsAndOutputs.hasSingleInput()
-                                   ? inputsAndOutputs.getFilenameOfFirstInput()
-                                   : StringRef();
+  const std::string firstInput =
```

Comment:
Good suggestion! I like it because `firstInput` gets passed into a constructor for `OutputFilesComputer`, and that class must take ownership of the information anyway.
So I think there's no concern about leaky invariants here. Pure win!
I'll change it to:   
`const StringRef firstInput =
      inputsAndOutputs.hasSingleInput()
          ? StringRef(inputsAndOutputs.getFilenameOfFirstInput())
          : StringRef();`.